client/core/core.go

```diff
@@ -2762,22 +2835,43 @@ func handleMatchRoute(c *Core, dc *dexConnection, msg *msgjson.Message) error {
 	if err != nil {
 		return fmt.Errorf("match request parsing error: %v", err)
 	}
+
+	// Acknowledgements MUST be in the same orders as the msgjson.Matches.
 	matches, acks, err := dc.parseMatches(msgMatches, true)
 	if err != nil {
+		// Even one failed match fails them all since the server requires acks
+		// for them all, and in the same order. TODO: consider lifting this
+		// requirement, which requires changes to the server's handling.
```

Comment:
This would not be a difficult change on the server, but right now Swapper.Negotiate requires it.